Modules/Keychain/Sources/Secured.swift

```diff
@@ -0,0 +1,173 @@
+import Foundation
+
+///
+/// `Secured` is a property wrapper that can be applied to any codable property to store it in the keychain
+///
+/// Example:
+/// ```swift
+/// @Secured(key: "keychain.value_key")
+/// var value: Bool
+/// ```
+@propertyWrapper
+public struct Secured<Value: Codable> {
+    private enum KeychainError: Error, LocalizedError {
+        case itemNotFound
+        case accessError(status: String)
+        case unexpectedData
+        case decodingError(error: Error)
+        case encodingError(error: Error)
+        case saveItemToKeychain(status: String)
+        case deleteItem
+
+        var errorDescription: String? {
+            switch self {
+            case .itemNotFound: return "The requests item could not be found in the keychain"
+            case .accessError: return "The keychain item could not be accessed"
+            case .unexpectedData: return "The retrieved keychain data type is unexpected"
```

Comment:
Can we somehow find out which data type was received and which one was expected? So we can enhance this error message?

Reply:
I already encapsulate the error thrown by JSONEncoder/-Decoder. The `unexpectedData` error wasn't used so I removed it